### PR TITLE
Support encoding proplists

### LINKED ISSUE #15: Support for proplist encoding
This is a question as much as an issue report. We are trying to swap JSX for Thoas in RabbitMQ
and observed that Thoas does not encode proplists, even though the input type spec suggests that it should:

``` erl
thoas:encode(#{a => 1}).
```

succeeds but

``` erl
thoas:encode(maps:to_list(#{a => 1})).
```

fails with a function clause. Is this limitation by design?

### PR DESCRIPTION
**Pull request by the-mikedavis**
Resolves #15

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,6 +71,7 @@ Erlang                        | JSON
 `#{}`                         | `{}`
 `#{<<"name">> => <<"Lucy">>}` | `{"name":"Lucy"}`
 `#{score => 42, win => true}` | `{"score":42,"win":true}`
+`{proplists, true}`           | `{"proplists":true}`
```
Comment (lpil) [inline README.md]: I think this should be this?

```suggestion
`[{proplists, true}]`           | `{"proplists":true}`
```

**Review by lpil — COMMENTED** [review]
Brilliant, thank you!

Left 1 question. Would you mind also updating the changelog?

**Comment by the-mikedavis** [comment]
Sure thing 👍

Would this be a `v0.3.0` release? I could just make an `UNRELEASED` section alternatively

**Comment by lpil** [comment]
Either would be fine thanks 👍

**Review by lpil — APPROVED** [review]
perfect, thanks!!!

**Pull request merged**